### PR TITLE
Flake8

### LINKED ISSUE #58: Enable flake8 tests and include in travis-ci configuration

### PR DESCRIPTION
**Pull request by AlanCoding**
Some features to resolve #58, flake8 and travis integration. One flake8 error still pops up, that 'file' is an undefined name. This syntax is probably antiquated somewhat, and it's difficult to understand what it should be, without breaking something else. For now, it has the --exit-zero flag set, indicating that passing the tests are not mandatory, but this is something up for discussion.

This is not merged with the other pull request. They both branch of master and will need to be merged later.

### CODE REVIEW

File: tests/compat.py [hunk 1]
```diff
@@ -34,3 +34,7 @@
     from io import TextIOWrapper
 except ImportError:  # Python < 3
     TextIOWrapper = file
```
Comment (AlanCoding) [inline tests/compat.py]: @jlaska This is the one line that I couldn't could to pass flake8. Here, 'file' is a keyword native to Python... but only for older versions. So I guess it's not surprising that flake8 does not see it as such. But then we're in a bit of a catch 22. How do we keep it functioning correctly and also pass flake8?

Comment (jlaska) [inline tests/compat.py]: `# NOQA` ?

Comment (AlanCoding) [inline tests/compat.py]: @jlaska Yep, noqa was what I was grasping for. Edited things to use that approach with recent commit. I'll give it a more thorough review later, but that plugs all the major holes.

**Comment by AlanCoding** [comment]
Coveralls seem to be complaining about a decreased coverage percent. I'm not sure, but I think this is because its baseline is the other branch (which does have higher coverage). Either way, I anticipate that will be fixed with a merge.

**Comment by jlaska** [comment]
Bonus, well done @AlanCoding .  The only comment I have regarding the use of `noqa` is that I'd probably prefer seeing that used on a per-line basis, rather than a per-file basis.  If you're okay with that, I'd proceed with merging afterwards.

**Comment by jlaska** [comment]
@AlanCoding looks great, go ahead and merge whenever ready.  Thanks for your contributions :)

**Pull request merged**